databuilder/utils/date_utils.py

```diff
@@ -38,6 +38,10 @@ def date_add_days(date, num_days):
     return date + datetime.timedelta(days=num_days)
 
 
+def valid_year(year):
+    return 0 < year <= 9999
```

Comment:
I wonder if it would be cleaner to change this to:
```py
def assert_valid_year(year):
    if not 0 < year <= 9999:
        raise ValueError(f"year {year} is out of range")
```
And then call this before entering the `try` block.